### PR TITLE
Listen for msg from specific url and with specific type

### PR DESCRIPTION
**Pull request by sharma66mahesh**

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -1,14 +1,19 @@
+import { useEffect } from "react"
 import sdk from "./sdk"
 
 function App() {
 
+  useEffect(() => {
+    sdk.listen(window, 'http://localhost:3000');
```
Comment (unijad) [inline src/App.tsx:7]: let's keep sdk listen, inside sdk initalization, no need for duplicates

File: src/login.tsx [hunk 2]
```diff
@@ -8,6 +8,10 @@ export default function Login() {
   const [isLoggingIn, setIsLoggingIn] = useState(false)
   const history = useHistory()
 
+  useEffect(() => {
+    sdk.listen(window, 'http://localhost:3000');
```
Comment (unijad) [inline src/login.tsx:12]: check other comment regarding duplicate listeners

File: src/magic.ts [hunk 3]
```diff
@@ -1,3 +1,11 @@
 import { Magic } from 'magic-sdk'
 
-export const magic = new Magic(process.env.REACT_APP_MAGIC_PUBLISHABLE_KEY || '')
\ No newline at end of file
+export const magic = new Magic(process.env.REACT_APP_MAGIC_PUBLISHABLE_KEY || '')
+
+// declare global {
```
Comment (unijad) [inline src/magic.ts]: remove commented out code

File: src/sdk.tsx [hunk 4]
```diff
@@ -7,19 +7,30 @@ class SimpleSDK {
     origin: string = window.location.origin;
 
     constructor() {
-        this.listen(window);
+        // this.listen(window);
```
Comment (unijad) [inline src/sdk.tsx]: remove commented code

File: src/sdk.tsx [hunk 5]
```diff
@@ -7,19 +7,30 @@ class SimpleSDK {
...
     }
 
     post(postEvent: postEvent, target: Window, origin: string = window.location.origin) {
         target.postMessage(postEvent, origin)
     }
 
-    listen(target: Window) {
+    listen(target: Window, senderUrl: string) {
```
Comment (unijad) [inline src/sdk.tsx:13]: we should have this configurable, to avoid xss requests

**Review by unijad — CHANGES_REQUESTED** [review]
check duplicates, and url configuration

File: src/App.tsx (hunk 1 above)
Comment (sharma66mahesh) [reply src/App.tsx:7]: In practice, there will be two listeners. One on the parent page, and one on the iframe. 
In our case, the `<App />` component is the parent and the `<Login />` is from the iframe. Hence the two separate listeners

File: src/sdk.tsx (hunk 5 above)
Comment (sharma66mahesh) [reply src/sdk.tsx:13]: Sorry, how do you want to set this up?

**Comment by Git-on-my-level** [comment]
Not sure why this is still open

**Pull request closed without merging**